test/backupMappings.js

```diff
@@ -0,0 +1,159 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* global afterEach beforeEach describe it */
+'use strict';
+
+const assert = require('node:assert');
+const request = require('../includes/request.js');
+const { backupBatchToBackupFileLine, backupBatchToLogFileLine, getPendingToFetchedMapper, logLineToBackupBatch, logLineToMetadata } = require('../includes/backupMappings.js');
+
+function assertFileLine(fileLine, expectedContent) {
+  assert.ok(fileLine, 'There should be a file line.');
+  assert.equal(typeof fileLine, 'string', 'The file line should be a string.');
+  assert.equal(fileLine, expectedContent, 'The file line should have the correct content.');
+}
+
+function assertBackupBatchObject(backupBatch, expectedCommand, expectedBatch, expectedDocs) {
+  assert.ok(backupBatch, 'The mapped value should be truthy.');
+  assert.strictEqual(typeof backupBatch, 'object');
+  assert.strictEqual(backupBatch.command, expectedCommand);
+  assert.strictEqual(backupBatch.batch, expectedBatch);
+  assert.deepStrictEqual(backupBatch.docs, expectedDocs);
+}
+
+describe('#unit backup mappings', function() {
+  const backupBatchTodo = { command: 't', batch: 0, docs: [{ id: 'doc1' }, { id: 'doc2' }] };
+  const backupBatchDone = { command: 't', batch: 0, docs: [{ _id: 'doc1', hello: 'world' }, { _id: 'doc2', foo: 'bar' }] };
+
+  describe('backupBatchToBackupFileLine', function() {
+    it('should correctly map to a backup line', function() {
+      const fileLine = backupBatchToBackupFileLine(backupBatchDone);
+      assertFileLine(fileLine, `${JSON.stringify(backupBatchDone.docs)}\n`);
+    });
+  });
+
+  describe('backupBatchToLogFileLine', function() {
+    it('should correctly map to a log file line', function() {
+      const fileLine = backupBatchToLogFileLine(backupBatchDone);
+      assertFileLine(fileLine, ':d batch0\n');
+    });
+  });
+
+  describe('log line mappers', function() {
+    const makeTestForLogLine = (fn, logLine, expected) => {
+      return function() {
+        const backupBatch = fn(logLine);
+        if (fn.name === logLineToMetadata.name) {
+          // In the metadata only cases we expect no docs
+          expected[2] = [];
+        }
+        assertBackupBatchObject(backupBatch, ...expected);
+      };
+    };
+
+    [logLineToBackupBatch, logLineToMetadata].forEach((fn) => {
+      describe(`${fn.name}`, function() {
+        it('should correctly map a todo log file line (static)',
+          makeTestForLogLine(fn,
+            ':t batch42 [{"id": "doc1"}, {"id": "doc2"}]',
```

Comment:
```suggestion
            `:t batch42 ${JSON.stringify(backupBatchTodo.docs)]`,
```

Reply:
So that is what happens in the "dynamic" version of the same test. I added a static version mainly for the purposes of differentiating between the code is broken vs the test is broken. Arguably in that case the assertion should also be static, so I've done that as well as making it more obviuosly different in 2319b897d76c9eb1dc3e3291eb3832eecbed75f4
Whilst it was useful during development, perhaps for regression purposes we might only want the dynamic version of the test. WDYT?

Reply:
This looks great! Thanks for the change!